### PR TITLE
fix(EMS-2207): GitHub Actions workflow cleanup

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction ✏️ 
Ensure GHA workflow are updated and correct enviornment variables are being fetched when being deployed to their respective enviornments.

## Resolution ✔️ 
* Removed frivolous variables from both or respective containers.
* Updated target environment from `prod` to `production` for naming consistency.
* `API_URL` container name correction.
* Container environment variables will be set during deployment.

## Miscellaneous ➕ 
* Dependencies updates

### CODE REVIEW

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -54,34 +54,6 @@ jobs:
...
-      - name: Defaults ✨
-        uses: Azure/cli@v1.0.8
-        with:
-          inlineScript: |
-            # Basic
-            az configure --defaults location=${{ vars.REGION }}
-            az configure --defaults group=rg-${{ env.PRODUCT }}-${{ github.ref_name }}-${{ vars.VERSION }}
-
-    #TODO: EMS-1176: MS SQL DB setup
-
```
Comment (ttbarnes) [inline .github/workflows/deployment.yml:84]: Is this removal intentional?

File: .github/workflows/infrastructure.yml [hunk 2]
```diff
@@ -554,13 +551,8 @@ jobs:
             WEBSITES_PORT='${{ vars.UI_PORT }}' \
             TLS_CERTIFICATE='${{ secrets.TLS_CERTIFICATE }}' \
             TLS_KEY='${{ secrets.TLS_KEY }}' \
-            WEBSITE_DNS_SERVER='${{ vars.WEBSITE_DNS_SERVER }}' \
-            WEBSITE_VNET_ROUTE_ALL=1 \
             SESSION_SECRET='${{ secrets.SESSION_SECRET }}' \
             GOOGLE_ANALYTICS_ID='${{ secrets.GOOGLE_ANALYTICS_ID }}' \
-            GOV_NOTIFY_API_KEY='${{ secrets.GOV_NOTIFY_API_KEY }}' \
-            COMPANIES_HOUSE_API_URL='${{ secrets.COMPANIES_HOUSE_API_URL }}' \
-            COMPANIES_HOUSE_API_KEY='${{ secrets.COMPANIES_HOUSE_API_KEY }}' \
```
Comment (ttbarnes) [inline .github/workflows/infrastructure.yml:563]: Why has `COMPANIES_HOUSE` been removed here?

File: .github/workflows/deployment.yml (hunk 1 above)
Comment (abhi-markan) [reply .github/workflows/deployment.yml:84]: Yes, we will be deploying straight to AMI.

File: .github/workflows/infrastructure.yml (hunk 2 above)
Comment (abhi-markan) [reply .github/workflows/infrastructure.yml:563]: Removed from `UI` container.

**Review by Zainzzkk — APPROVED** [review]

**Review by ttbarnes — APPROVED** [review]
